### ISSUE TITLE
Tests for functions simplecycles, simplecyclescount, maxsimplecycles should test for different eltypes

### ISSUE THREAD

**Issue by simonschoelly** [JuliaGraphs/LightGraphs.jl#1146]
Currently the tests for the functions defined in `src/cycles/johnson.jl` do only test for graphs of eltype `Int`. We should loop over different eltypes and also test for these.
For that we have two helper functions `testgraphs` and `testdigraphs` that convert a graph to suitable eltypes that we want to test. For an example how that can be done, one can look at the tests defined in `test/cycles/karp.jl`

The tests for `johnson.jl` can be found in the file `test/cycles/johnson.jl`

**Comment by ChamanAgrawal** [comment]
I am working on this.

**Comment by ChamanAgrawal** [comment]
On trying this 
```
for g in testdigraphs(pathdg)
        @test maxsimplecycles(g) == 0
        @test maxsimplecycles(g, false) == 84
        @test simplecyclescount(g) == 0
        @test length(simplecycles(g)) == 0
        @test isempty(simplecycles(g)) == true
        @test isempty(simplecycles_iter(g)) == true
        @test simplecycleslength(g) == (zeros(5), 0)
        @test simplecyclescount(g, 10) == 0
        @test isempty(simplecycles_iter(g, 10)) == true
        @test simplecycleslength(g, 10) == (zeros(5), 0)
    end
```
It gives error on all tests. The error on the first few is
```
Cycles: Error During Test at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:12
  Test threw exception
  Expression: simplecyclescount(g) == 0
  MethodError: no method matching circuit_iter(::Int64, ::SimpleDiGraph{UInt8}, ::LightGraphs.JohnsonVisitor{UInt8}, ::Array{UInt8,1}, ::Channel{Any})
  Closest candidates are:
    circuit_iter(::T<:Integer, ::##480, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, !Matched::Array{T<:Integer,1}, ::Channel) where {T<:Integer, ##480} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350
    circuit_iter(::T<:Integer, ::##480, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, !Matched::Array{T<:Integer,1}, ::Channel, !Matched::T<:Integer) where {T<:Integer, ##480} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350
    circuit_iter(!Matched::Type{IsDirected{##480}}, !Matched::T<:Integer, ::##480, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, !Matched::Array{T<:Integer,1}, !Matched::Channel, !Matched::T<:Integer) where {T<:Integer, ##480} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:332
  Stacktrace:
   [1] check_channel_state at ./channels.jl:120 [inlined]
   [2] take_unbuffered(::Channel{Any}) at ./channels.jl:318
   [3] take! at ./channels.jl:306 [inlined]
   [4] iterate(::Channel{Any}, ::Nothing) at ./channels.jl:386
   [5] iterate at ./channels.jl:385 [inlined]
   [6] iterate at ./iterators.jl:591 [inlined]
   [7] iterate at ./iterators.jl:589 [inlined]
   [8] macro expansion at /home/chaman/git/LightGraphs.jl/src/cycles/johnson.jl:304 [inlined]
   [9] simplecyclescount(::Type{IsDirected{SimpleDiGraph{UInt8}}}, ::SimpleDiGraph{UInt8}, ::Int64) at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:332
   [10] simplecyclescount at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350 [inlined] (repeats 2 times)
   [11] macro expansion at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:12 [inlined]
   [12] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
   [13] top-level scope at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:2
Cycles: Error During Test at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:13
  Test threw exception
  Expression: length(simplecycles(g)) == 0
  MethodError: no method matching circuit(::Int64, ::SimpleDiGraph{UInt8}, ::LightGraphs.JohnsonVisitor{UInt8}, ::Array{Array{Int64,1},1}, ::Array{UInt8,1})
  Closest candidates are:
    circuit(::T<:Integer, ::##476, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, ::Array{Array{T<:Integer,1},1}, !Matched::Array{T<:Integer,1}) where {T<:Integer, ##476} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350
    circuit(::T<:Integer, ::##476, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, ::Array{Array{T<:Integer,1},1}, !Matched::Array{T<:Integer,1}, !Matched::T<:Integer) where {T<:Integer, ##476} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350
    circuit(!Matched::Type{IsDirected{##476}}, !Matched::T<:Integer, ::##476, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, !Matched::Array{Array{T<:Integer,1},1}, !Matched::Array{T<:Integer,1}, !Matched::T<:Integer) where {T<:Integer, ##476} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:332
  Stacktrace:
   [1] macro expansion at /home/chaman/git/LightGraphs.jl/src/cycles/johnson.jl:192 [inlined]
   [2] simplecycles(::Type{IsDirected{SimpleDiGraph{UInt8}}}, ::SimpleDiGraph{UInt8}) at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:332
   [3] simplecycles(::SimpleDiGraph{UInt8}) at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350
   [4] macro expansion at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:13 [inlined]
   [5] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
   [6] top-level scope at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:2
Cycles: Error During Test at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:14
  Test threw exception
  Expression: isempty(simplecycles(g)) == true
  MethodError: no method matching circuit(::Int64, ::SimpleDiGraph{UInt8}, ::LightGraphs.JohnsonVisitor{UInt8}, ::Array{Array{Int64,1},1}, ::Array{UInt8,1})
  Closest candidates are:
    circuit(::T<:Integer, ::##476, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, ::Array{Array{T<:Integer,1},1}, !Matched::Array{T<:Integer,1}) where {T<:Integer, ##476} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350
    circuit(::T<:Integer, ::##476, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, ::Array{Array{T<:Integer,1},1}, !Matched::Array{T<:Integer,1}, !Matched::T<:Integer) where {T<:Integer, ##476} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350
    circuit(!Matched::Type{IsDirected{##476}}, !Matched::T<:Integer, ::##476, !Matched::LightGraphs.JohnsonVisitor{T<:Integer}, !Matched::Array{Array{T<:Integer,1},1}, !Matched::Array{T<:Integer,1}, !Matched::T<:Integer) where {T<:Integer, ##476} at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:332
  Stacktrace:
   [1] macro expansion at /home/chaman/git/LightGraphs.jl/src/cycles/johnson.jl:192 [inlined]
   [2] simplecycles(::Type{IsDirected{SimpleDiGraph{UInt8}}}, ::SimpleDiGraph{UInt8}) at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:332
   [3] simplecycles(::SimpleDiGraph{UInt8}) at /home/chaman/.julia/packages/SimpleTraits/CZOA2/src/SimpleTraits.jl:350
   [4] macro expansion at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:14 [inlined]
   [5] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
   [6] top-level scope at /home/chaman/git/LightGraphs.jl/test/cycles/johnson.jl:2

```
Will this be because of the first argument to be of type Int64 ?

**Comment by abhinavmehndiratta** [comment]
This appears to be a bug.Try correcting
https://github.com/JuliaGraphs/LightGraphs.jl/blob/2df8575d03e502d778ef1eafb56db6d34b304c01/src/cycles/johnson.jl#L275
to `circuit_iter(eltype(wdg)(1), wdg, visitor, vmap, cycle)` and run the tests again.

**Comment by ChamanAgrawal** [comment]
Yeah I was thinking the same, just got time to get my hands dirty :)

**Comment by simonschoelly** [comment]
This was was solved by #1156, so I am closing this.
